docs(applicationset): fix layout matrix/merge generator restrictions

sboschman opened the pull request:
The applicationset docs for matrix/merge generators contain a numbered list of restrictions. All items are numbered '1', as the yaml examples reset the markdown numbering.

mkdocs does not support fenced code blocks inside lists, like github markdown does:
> Note that fenced code blocks can not be indented. Therefore, they cannot be nested inside list items, blockquotes, etc.

Using indented code blocks inside the list now, `make serve-docs-local` shows correctly numbered items and formatted code blocks. 

Note on DCO:

If the DCO action in the integration test fails, one or more of your commits are not signed off. Please click on the *Details* link next to the DCO action for instructions on how to resolve this.

Checklist:

* [X] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo-cd/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this does not need to be in the release notes.
* [ ] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] I've included "Closes [ISSUE #]" or "Fixes [ISSUE #]" in the description to automatically close the associated issue.
* [ ] I've updated both the CLI and UI to expose my feature, or I plan to submit a second PR with them.
* [X] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [ ] Optional. My organization is added to USERS.md.
* [X] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/tree/master/community#contributing-to-argo)
* [ ] I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.
* [ ] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/latest/developer-guide/ci/)).

crenshaw-dev reviewed (changes requested):
Thanks @sboschman! Unfortunately, this collapses the code blocks in mkdocs/readthedocs (tested with `make serve-docs-local`). Can you take a look at https://github.com/sboschman/argo-cd/pull/1 ?

sboschman commented:
Had to set strict=false in mkdocs.yml to get it working:

```
% make serve-docs-local        
mkdocs serve
INFO     -  Building documentation...
WARNING  -  Config value: 'google_analytics'. Warning: The configuration option google_analytics has been deprecated and will be removed in a future release of MkDocs. See the options available on your theme for
            an alternative.
Aborted with 1 Configuration Warnings in 'strict' mode!
```

sboschman commented:
@crenshaw-dev as the DCO failed on your commit, and 3 commits to fix some numbering in the docs on the main branch seems a bit excessive, I just squashed everything into a single commit and added your name as co-author to that single commit. Hopefully that is ok with you.

crenshaw-dev reviewed (approved):
lgtm, thanks @sboschman!
